python/cuml/common/base.pyx

```diff
@@ -59,8 +59,8 @@ class Base:
         algo = MyAlgo(handle=handle)
         algo.fit(...)
         result = algo.predict(...)
-        # final synchronization of all work launched/dependent on this stream
-        stream.sync()
+        # final sync of all gpu-work launched inside this object
+        base.sync()
```

Comment:
From the guarantees the C++ API make there is not functional difference between `base.sync()` and `stream.sync()`. So what is the motivation to introduce `base.sync()`? From my perspective not having `base.sync()` is more inline with how other CUDA libraries work and it might confuse users as they expect `base.sync()` to do more than just sync the stream.

Reply:
Here's the issue. `cuml.cuda.Stream` is an owning class. Meaning, it owns the underlying `cudaStream_t` and it creates a new stream everytime. However, the `cumlHandle::setStream` accepts only raw `cudaStream_t` objects. If you think we shouldn't have `base.sync`, then a cleaner approach, would be for `ML::cumlHandle` itself to provide a `sync` method. That can simplify this current "sync" proposal of mine. What say?

Reply:
@jirikraus like we discussed, I've removed `base.sync` and used `base.handle.sync` instead. Can you check now?

Reply:
I believe this can be resolved now. If not, please let me know.